app/controllers/UserProfileController.java

```diff
@@ -59,4 +70,20 @@ public Result updateUserProfile() throws Exception {
         
         return okResult("Profile updated.");
     }
+    
+    public Result createExternalIdentifier() throws Exception {
+        UserSession session = getAuthenticatedSession();
+
+        ExternalIdentifier externalId = ExternalIdentifier.fromJson(requestToJSON(request()));
+        
+        // TODO: An annotation-based validator would make these trivial validations 
+        // easier to factor out.
+        if (isBlank(externalId.getIdentifier())) {
```

Comment:
I agree with this but there's no business-specific service involved here. It's calling directly to the options service which throws an IllegalArgumentException if the identifier isn't submitted, not the invalid entity exception you should get if your JSON is wrong.